bin/ch/ch.cpp

```diff
@@ -671,6 +671,10 @@ HRESULT ExecuteTest(const char* fileName)
         }
     }
 Error:
+    if( fileContents != nullptr)
+    {
+        free((char*)fileContents);
```

Comment:
Each ExternalArrayBuffer can take a finalizeCallback argument and a callbackState argument. On finalize, the finalizeCallback, if it's non-null, is called on the callbackState. There are indeed a few cases where a finalizeCallback is specified in ch (as a lambda wrapper around free()). However, in the few cases where it is specified, the provided callbackState is nullptr. This means that it does not actually do anything. In the case of calling free, free(nullptr) is specified as doing nothing. This changeset already includes fixes to the cases in WScriptJsrt where ExternalArrayBuffer was just being called without finalizeCallbacks (=nullptr) and was leaking the source body of every script run. There's two locations in ch.cpp where the finalizeCallback was specified, but the callbackState is nullptr, and one in ch.cpp and one in Debugger.cpp where both are nullptr. Manual memory management of the style at the top of this comment chain is the reason that the non-leaking cases weren't leaking. There's a few situations in which this is actually complicated a bit; For instance, there's at least one path that creates multiple ExternalArrayBuffers for the same backing buffer.